src/hotspot/os/linux/os_linux.cpp

```diff
@@ -3366,6 +3366,9 @@ bool os::pd_create_stack_guard_pages(char* addr, size_t size) {
 
     if (mincore((address)stack_extent, os::vm_page_size(), vec) == -1) {
       // Fallback to slow path on all errors, including EAGAIN
+      assert((uintptr_t)addr >= stack_extent,
+             "Sanity: addr should be larger than extent, " PTR_FORMAT " >= " PTR_FORMAT,
+             p2i(addr), stack_extent);
```

Comment:
Should we not do something in non-debug builds too? Maybe guarantee() instead?

Reply:
Nah, I think `assert` is okay here. It is supposed to aid debugging, not crash hard.